openml/study/functions.py

```diff
@@ -53,7 +58,180 @@ def get_study(study_id, type=None):
 
     if 'oml:setups' in result_dict:
         setups = [int(x) for x in result_dict['oml:setups']['oml:setup_id']]
+    
+    if 'oml:runs' in result_dict:
+        runs = [int(x) for x in result_dict['oml:runs']['oml:run_id']]
 
-    study = OpenMLStudy(id, name, description, creation_date, creator, tags,
-                        datasets, tasks, flows, setups)
+    study = OpenMLStudy(
+            study_id=study_id,
+            alias=alias,
+            main_knowledge_type=main_knowledge_type,
+            benchmark_suite=benchmark_suite,
+            name=name, 
+            description=description, 
+            creation_date=creation_date, 
+            creator=creator, 
+            tags=tags,
+            data=datasets, 
+            tasks=tasks, 
+            flows=flows, 
+            setups=setups,
+            runs=runs
+    )
     return study
+
+
+def create_study(alias, benchmark_suite, name, description, run_ids):
+    """
+    Creates an OpenML study (collection of knowledge types, where the runs
+    are the linked entity)
+    
+    Parameters:
+    -----------
+    alias : str (optional)
+        a string ID, unique on server (url-friendly)
+    benchmark_suite : int (optional)
+        the benchmark suite (another study) upon which this study is ran.
+    name : str
+        the name of the study (meta-info)
+    description : str
+        brief description (meta-info)
+    run_ids : list
+        a list of run ids associated with this study
+    
+    Returns:
+    --------
+    OpenMLStudy
+        A local OpenML study object (call publish method to upload to server)
+    """
+    return OpenMLStudy(
+            study_id=None,
+            alias=alias,
+            main_knowledge_type='run',
+            benchmark_suite=benchmark_suite,
+            name=name, 
+            description=description,
+            creation_date=None,
+            creator=None,
+            tags=None,
+            data=None,
+            tasks=None,
+            flows=None,
+            setups=None,
+            runs=run_ids
+    )
+
+
+def create_benchmark_suite(alias, name, description, task_ids):
+    """
+    Creates an OpenML benchmark suite (collection of knowledge types, where 
+    the tasks are the linked entity)
+    
+    Parameters:
+    -----------
+    alias : str (optional)
+        a string ID, unique on server (url-friendly)
+    name : str
+        the name of the study (meta-info)
+    description : str
+        brief description (meta-info)
+    task_ids : list
+        a list of task ids associated with this study
+    
+    Returns:
+    --------
+    OpenMLStudy
+        A local OpenML study object (call publish method to upload to server)
+    """
+    return OpenMLStudy(
```

Comment:
Hm, I'm not sure about this one. Does the server already distinguish studies and suites? If yes, we should add a subclass `OpenMLSuite` to make the code easier to understand.

Reply:
The server does not distinguish between those. It is a parameter in the database (main knowledge type is either runs or tasks)

Reply:
I assumed that. But wouldn't it from a usage perspective make sense to have two separate objects for this?

Reply:
Alternatively we could have 2 objects that both have a different contructor. However i figured it might be complicated because since the server always returns a "study" tag and we have to differentiate. Seems like more potential legacy and maintenance to me, but all together it doesn't make a great difference

Reply:
Okay, let's then leave it like it is and see how things are going.